src/Transformer/CustomTransformer/CustomPropertyTransformerInterface.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AutoMapper\Transformer\CustomTransformer;
+
+use Symfony\Component\PropertyInfo\Type;
+
+interface CustomPropertyTransformerInterface extends CustomTransformerInterface
```

Comment:
> I would use something like $sourcePropertyName to not confuse things when target property name is different because of some NameConverter

This makes me raise two questions:
- shouldn't we somehow introduce the same mechanism with `targetPropertyName`?
-  currently we're actually passing the "source property" to the `transform()` method, I'm wondering if we shouldn't pass the whole `$source` object in order to be able to access to whole source object in the tranformation (ie: we could need `$source->year`, `$source->month`, `$source->day` in order to generate `$target->birthDate`

WDYT?